app/views/users/show.html.erb

```diff
@@ -1 +1,19 @@
-<p> show each user </p>
\ No newline at end of file
+<%# <p> show each user </p> %>
+
+<section>
+  <%= render 'layouts/user_card', user: @user %>
+  <div class="user">
+    <h3 class="bio">Bio</h3>
+    <h4 class="">
+      <%= @user.bio %>
+    </h4>
+  </div>
+  <div>
+    <% @user.posts.each do |post| %>
+      <%= render 'layouts/post_card', post: post %>
+    <% end %>
+  </div>
```

Comment:
- [ ] `@user.return_recent_posts` is missing in your user _show_ view. As this activity requires that we use the `recent_post` method defined in the _user_ model, in your case `return_recent_posts`, to render the user's 3 most recent post in the `app/views/users/show.html.erb` view, it should be similar to:

```ruby
# app/views/users/show.html.erb

[...]

<% @user.recent_post.each do |post| %>

<h4><%= link_to post.title, user_post_path(post) %></h4>
<p><%= post.text %></p>
<p>
<small>Comments: <%= post.comments_counter || 0 %>,</small> 
<small>Likes: <%= post.likes_counter || 0 %></small>
</p>

<% end %>

<button>
<%= link_to 'See all Post', user_posts_path(@user) %>
</button>
```